### PR TITLE
doc: add IOC virtualization HLD

### PR DESCRIPTION
**Pull request by dbkinder**
Convert and edit IOC virtualization doc to add to ACRN documentation.

Signed-off-by: David B. Kinder <david.b.kinder@intel.com>

### CODE REVIEW

File: doc/developer-guides/hld/hv-ioc-virt.rst [hunk 1]
```diff
@@ -0,0 +1,726 @@
...
+   -l com2,/run/acrn/ioc_$vm_name
+
+
+Porting and adaptation to different platforms
+*********************************************
+
+Reference Documents
+*******************
+
+- 519325_CBC_Protocol_Service_Interfaces_Specification_Rev1.2.pdf
```
Comment (dbkinder) [inline doc/developer-guides/hld/hv-ioc-virt.rst]: @liuyuan-GitHub Can you provide links to these (public) documents

File: doc/developer-guides/hld/hld-hypervisor.rst [hunk 2]
```diff
@@ -12,6 +12,7 @@ Hypervisor high-level design
    Memory management <hv-memmgt>
    I/O Emulation <hv-io-emulation>
    Physical Interrupt <hv-interrupt>
+   IOC Mediator <hv-ioc-virt>
```
Comment (dbkinder) [inline doc/developer-guides/hld/hld-hypervisor.rst]: @liuyuan-GitHub I took a guess where to put this in the HLD document set... let me know if there's a better place.

File: doc/developer-guides/hld/hv-ioc-virt.rst (hunk 1 above)
Comment (liuyuan-GitHub) [reply doc/developer-guides/hld/hv-ioc-virt.rst]: These reference documents are in my local, do we have a public shared directory for these documents?

File: doc/developer-guides/hld/hld-hypervisor.rst (hunk 2 above)
Comment (liuyuan-GitHub) [reply doc/developer-guides/hld/hld-hypervisor.rst]: The location is fine.
By the way, the IOC mediator only work on MRB platform for IVI, the UP2 and NUC do not support IOC hardware and IOC is disabled by default on their boot script.

File: doc/developer-guides/hld/hv-ioc-virt.rst (hunk 1 above)
Comment (liuyuan-GitHub) [reply doc/developer-guides/hld/hv-ioc-virt.rst]: Please remove reference documents

File: doc/developer-guides/hld/hv-ioc-virt.rst [hunk 3]
```diff
@@ -0,0 +1,726 @@
...
+- 566739_CBC_Specification_ServiceInterfaces_KC3.0_Rev0.2.pdf
```
Comment (liuyuan-GitHub) [inline doc/developer-guides/hld/hv-ioc-virt.rst]: Please remove reference document

File: doc/developer-guides/hld/hv-ioc-virt.rst [hunk 4]
```diff
@@ -0,0 +1,726 @@
...
+- 519208_CBC_Spec_CommunicationsLayer_Rev1.0.pdf
```
Comment (liuyuan-GitHub) [inline doc/developer-guides/hld/hv-ioc-virt.rst]: Please remove reference documents

File: doc/developer-guides/hld/hld-hypervisor.rst (hunk 2 above)
Comment (hwang37) [reply doc/developer-guides/hld/hld-hypervisor.rst]: Agree Dave.
We can put it under Developer Guide->Developer Primer, along with Memory Management, I/O Emulation, etc.

**Pull request merged**